gb-cpu/src/processor.rs

```diff
@@ -0,0 +1,15 @@
+use super::cpu::Cpu;
+use crate::memory::Memory;
+
+#[derive(Debug)]
+pub struct Processor {
+    clock: u32,
+    cpu: Cpu,
+    memory: Memory,
+}
+
+impl Processor {
+    pub fn run(&mut self) {
```

Comment:
Bad naming for method run.
By only reading the method, I expect the Processor to run in a loop not having to call run again and again to tick the CPU

Reply:
Totally agree, I wasn't sure so i just used the first name that came to my mind.

But that's exactly what it will eventually do, but i like to put the loop when every that runs inside is ready!

Reply:
We can put the CPU into it's own loop, remember we have to run the PPU in parralel

Reply:
Sure, it's a draft, so i'm really open for discution. It's a proposition, nothing is set in stone, let's figure it out